src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,35 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executor.submit(() -> {
+                try {
+                    listener.onEvent(event);
+                } catch (Exception e) {
+                    throw new RuntimeException("Failed to handle an event" + event, e);
```

Comment:
Throwing a RuntimeException inside the lambda passed to 'submit' will not halt the program or the executor service. It will only log the exception to the console, which might not be the desired behavior. Consider logging the exception using a logging framework or handling it in another way that aligns with your application's error-handling strategy.